File: app/schemas/job.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class JobCreate(BaseModel):
    title: str = Field(max_length=150)
    description: str = Field(max_length=5000)
    location: str | None = Field(default=None, max_length=100)
    job_type: Literal["onsite", "remote", "hybrid"] = "onsite"


class JobStatusUpdate(BaseModel):
    status: Literal["open", "closed"]


class CompanyOut(BaseModel):
    id: uuid.UUID
    name: str

    class Config:
        from_attributes = True


class JobOut(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    location: str | None
    job_type: str
    status: str
    company: CompanyOut
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/job.ts
export type JobType = 'onsite' | 'remote' | 'hybrid'
export type JobStatus = 'open' | 'closed'

export interface JobCreate {
  title: string
  description: string
  location?: string | null
  job_type?: JobType
}

export interface JobStatusUpdate {
  status: JobStatus
}

export interface CompanyOut {
  id: string
  name: string
}

export interface JobOut {
  id: string
  title: string
  description: string
  location: string | null
  job_type: JobType
  status: JobStatus
  company: CompanyOut
  created_at: string
}

export interface JobListParams {
  page?: number
  page_size?: number
  job_type?: JobType
  location?: string
}
